File: estudos/js/crud_memoria.js
let tarefas = [];
let proximoId = 1;

function criarTarefa(titulo, descricao) {
  const tarefa = {
    id: proximoId++,
    titulo,
    descricao,
    concluida: false,
  };
  tarefas.push(tarefa);
  return tarefa;
}

function listarTarefas() {
  return tarefas;
}

function atualizarTarefa(id, dados) {
  const tarefa = tarefas.find(t => t.id === id);
  if (tarefa) {
    if (dados.titulo !== undefined) {
      tarefa.titulo = dados.titulo;
    }
    if (dados.descricao !== undefined) {
      tarefa.descricao = dados.descricao;
    }
    if (dados.concluida !== undefined) {
      tarefa.concluida = dados.concluida;
    }
    return tarefa;
  }
}

function deletarTarefa(id) {
  tarefas = tarefas.filter(t => t.id !== id);
}

// Testando as funções
console.log("Criando tarefas...");
criarTarefa("Estudar JavaScript", "Focar em CRUD");
criarTarefa("Ler documentação", "MDN e outras fontes");

console.log("\nTarefas:");
console.log(listarTarefas());

console.log("\nAtualizando tarefa 1...");
atualizarTarefa(1, { concluida: true });

console.log("\nTarefas após atualização:");
console.log(listarTarefas());

console.log("\nDeletando tarefa 2...");
deletarTarefa(2);

console.log("\nTarefas finais:");
console.log(listarTarefas());

File: estudos/python/crud_memoria.py
tarefas = []
proximo_id = 1

def criar_tarefa(titulo, descricao):
  global proximo_id
  tarefa = {
    'id': proximo_id,
    'titulo': titulo,
    'descricao': descricao,
    'concluida': False
  }
  tarefas.append(tarefa)
  proximo_id += 1
  return tarefa

def listar_tarefas():
  return tarefas

def atualizar_tarefa(id, titulo=None, descricao=None, concluida=None):
  for tarefa in tarefas:
    if tarefa['id'] == id:
      if titulo is not None:
        tarefa['titulo'] = titulo
      if descricao is not None:
        tarefa['descricao'] = descricao
      if concluida is not None:
        tarefa['concluida'] = concluida
      return tarefa
  return None

def deletar_tarefa(id_tarefa):
  global tarefas
  tarefas = [t for t in tarefas if t['id'] != id_tarefa]
  return True

#Testando as funções
if __name__ == "__main__":
  print("Criando tarefas...")
  criar_tarefa("Estudar Python", "Focar no CRUD")
  criar_tarefa("Ler artigo", "Ler artigo sobre APIs")

  print("\nTarefas:")
  print(listar_tarefas())

  print("\nAtualizando a primeira tarefa...")
  atualizar_tarefa(1, concluida=True)

  print("\nTarefas atualizadas:")
  print(listar_tarefas())

  print("\nDeletando a segunda tarefa...")
  deletar_tarefa(2)

  print("\nTarefas finais:")
  print(listar_tarefas())